src/EventStore.Core/Services/SubscriptionMetric.cs

```diff
@@ -0,0 +1,74 @@
+#nullable enable
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+using EventStore.Core.Messages;
+
+namespace EventStore.Core.Services;
+
+public class SubscriptionMetric {
+	private readonly ConcurrentDictionary<Guid, MonitoringMessage.SubscriptionStats> _subscriptionStats;
+
+	public SubscriptionMetric(Meter meter, string name) {
+		_subscriptionStats = new();
+
+		meter.CreateObservableUpDownCounter($"{name}-completed", Completed);
+		meter.CreateObservableUpDownCounter($"{name}-subscription-position", SubscriptionPosition);
+		meter.CreateObservableUpDownCounter($"{name}-stream-position", StreamPosition);
+	}
+
+	public void Add(Guid subscriptionId, string? streamName, long endOfStream) =>
+		_subscriptionStats.TryAdd(subscriptionId, new(subscriptionId, streamName, EndOfStream: endOfStream));
+
+	public void Remove(Guid subscriptionId) => _subscriptionStats.TryRemove(subscriptionId, out _);
```

Comment:
could you double check what happens in the dashboard if we stop reporting a particular subscription, does it stop drawing it or does it just keep drawing the last reported values for a while